website/server/models/user/methods.js

```diff
@@ -20,6 +20,49 @@ schema.methods.getGroups = function getUserGroups () {
   return userGroups;
 };
 
+// Get an array of error message keys that would be thrown if the given interaction was attempted
+schema.methods.getObjectionsToInteractionIfAny = function getObjectionsToInteractionIfAny (interaction, receiver) {
+  let sender = this;
+
+  /* eslint-disable no-unused-vars */
+  let checks = {
+    always: [
+      // Revoked chat privileges block all interactions to prevent the evading of harassment protections
+      // See issue #7971 for some discussion
+      (sndr, rcvr) => sndr.flags.chatRevoked && 'chatPrivilegesRevoked',
+
+      // Direct user blocks prevent all interactions
+      (sndr, rcvr) => rcvr.inbox.blocks.includes(sndr._id) && 'notAuthorizedToSendMessageToThisUser',
+      (sndr, rcvr) => sndr.inbox.blocks.includes(rcvr._id) && 'notAuthorizedToSendMessageToThisUser',
+    ],
+
+    'send-private-message': [
+      // Private messaging has an opt-out, which does not affect other interactions
+      (sndr, rcvr) => rcvr.inbox.optOut && 'notAuthorizedToSendMessageToThisUser',
+
+      // NB: We allow a player to message themselves so they can test how PMs work or send their own notes to themselves
```

Comment:
Actually, it's not your fault, the NB wasn't in your original line. I added it because the comment looked a bit out of place to me without any code to follow it. I didn't consider it might confuse people, so I agree it's better to remove it. :)